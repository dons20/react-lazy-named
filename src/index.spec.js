import React, { Suspense } from 'react';
import { create } from 'react-test-renderer';
import lazy from './index';

function Text(props) {
  return props.text;
}

function delay(ms) {
  return new Promise(resolve => setTimeout(() => resolve(), ms));
}

async function fakeImport(result) {
  return { default: result };
}

describe('useWhyChange', () => {
  it('stays quiet if updates never happen', async () => {
    const LazyText = lazy(() => fakeImport(Text));

    const root = create(
      <Suspense fallback={<Text text="Loading..." />}>
        <LazyText text="Hi" />
      </Suspense>
    );

    expect(root.toJSON()).toMatchSnapshot();

    await delay(10);

    expect(root.toJSON()).toMatchSnapshot();
  });
});
