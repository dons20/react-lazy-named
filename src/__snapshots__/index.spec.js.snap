// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useWhyChange stays quiet if updates never happen 1`] = `"Loading..."`;

exports[`useWhyChange stays quiet if updates never happen 2`] = `"Hi"`;
